### DISCUSSION TITLE
How can I stop my github action being fired 3 times when I create a new release?

### DISCUSSION THREAD

**Discussion by baltermia** [community/community#25656] [Actions]
<p>This is my first post in this community, if I do something wrong kindly tell me please.</p>
<hr>
<p>To the issue:<br>
I have a action that is fired each time I create a new release in my repository. The code for that looks like this:</p>
<pre><code class="lang-auto">on:
  release:
  workflow_dispatch:
</code></pre>
<p>I also want to be able to fire the action manually, that is why I also have the <code>workflow_dispatch</code> in there.</p>
<p>Now the action does what it should - it fires as soon as I create a new release. The problem is, that it fires the action three times at the exact same time. I only create 1 release, the action is fired three times.</p>
<p>You can see the entire code for the action <a href="https://github.com/speyck/BarcodeReader/blob/36e1ef8518634c3e4cfb93fa20104f11ac21da53/.github/workflows/package_on_release.yml" rel="noopener nofollow ugc">here</a>.</p>
<p>Is there any code I could add to solve this issue?</p>

**Comment by Simran-B** [comment, marked as answer]
<p><code>on: release</code> triggers on all event types by default. You can limit it to e.g. <code>published</code>:</p>
<pre><code class="lang-yaml">on:
  release:
    types: [published]
</code></pre>
<p>Also see the docs: <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release" class="inline-onebox" rel="noopener nofollow ugc">Events that trigger workflows - GitHub Docs</a></p>

**Comment by baltermia** [comment]
<p>Thank you very much, that is exactly what I needed.</p>
